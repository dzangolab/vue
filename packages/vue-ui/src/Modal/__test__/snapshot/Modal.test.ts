import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import Modal from "../../Index.vue";

describe("Modal", () => {
  it("matches snapshot", () => {
    const wrapper = mount(Modal, {
      props: {
        show: true,
        title: "Test",
      },
      slots: {
        default: "<p>This is test content</p>",
        footer: "<p>Modal footer</p>",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
