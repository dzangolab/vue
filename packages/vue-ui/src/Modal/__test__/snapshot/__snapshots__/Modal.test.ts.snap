// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > matches snapshot 1`] = `
<div
  class="modal"
>
  <div
    class="overlay"
  >
    <div
      class="dialog active"
    >
      <div
        class="header"
      >
        
        <span
          role="heading"
        >
          Test
        </span>
        
        <svg
          aria-label="close modal"
          height="1.5rem"
          tabindex="0"
          viewBox="0 0 16 16"
          width="1.5rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="content"
      >
        
        <p>
          This is test content
        </p>
        
      </div>
      <div
        class="footer"
      >
        
        <p>
          Modal footer
        </p>
        
      </div>
    </div>
  </div>
</div>
`;
